piquasso/_simulators/fock/general/state.py

```diff
@@ -318,3 +319,281 @@ def get_purity(self):
+        y_points: int = 200,
+        library: Literal["matplotlib", "plotly"] = "matplotlib",
+    ):
+        r"""
+        Plot the single‐mode Wigner function of a (mixed) FockState by numerical integration.
+
+        This uses the formula
+            W(x,p) = (1 / (π ℏ)) ∫_{-y_max}^{+y_max} e^{(2 i p y)/ℏ} 
+                      ⟨ x − y | ρ | x + y ⟩ dy,
+        where ρ is the single‐mode density matrix in Fock basis, and
+            ⟨q | n⟩ = φ_n(q) = (π ℏ)^(-1/4) (1/√(2^n n!)) H_n(q/√ℏ) e^(−q^2/(2ℏ)).
+
+        Steps:
+          1. Reduce the full `FockState` to obtain the single‐mode density matrix ρ (shape = cutoff × cutoff).
+          2. Build a uniform grid in x and p (each axis of length = resolution).
+          3. Build a uniform grid of y∈[−y_integration_range, +y_integration_range] (length = y_points).
+          4. Precompute φ_n(x_i ± y_k) for n=0..cutoff−1, i=0..resolution−1, k=0..y_points−1.
+          5. For each y_k, form the matrix
+               M_k[i,j] = sum_{m,n=0}^{cutoff−1}  ρ[m,n] · φ_m(x_i − y_k) · conj[φ_n(x_i + y_k)].
+          6. Multiply M_k[i,j] by exp(2 i p_j y_k / ℏ), integrate over y via trapezoidal rule,
+             and multiply by (1/(π ℏ)) ⇒ W[i,j].
+          7. Defer the import of Matplotlib or Plotly until plotting. If missing, raise ImportError.
+
+        Parameters
+        ----------
+        mode : int, default=0
+            Which mode index (0‐based) to visualize. Must satisfy 0 ≤ mode < state.d.
+        xlims : (float, float) or None
+            If provided, these are (x_min, x_max). Otherwise, defaults to ±4 σ_x, where
+            σ_x = √⟨x²⟩ for that single mode (computed from its reduced density).
+        plims : (float, float) or None
+            If provided, these are (p_min, p_max). Otherwise, defaults to ±4 σ_p.
+        resolution : int, default=200
+            Number of grid points along *each* of the x and p axes
+            (so the total grid is resolution×resolution).
+        y_integration_range : float, default=5.0
+            We truncate the y‐integral to [−y_range, +y_range]. If ℏ is small,
+            you may need a larger range; if it's big or the state is narrow,
+            you can choose smaller.
+        y_points : int, default=200
+            Number of points to discretize the y‐integral on [−y_range, +y_range].
+        library : {"matplotlib", "plotly"}
+            Which plotting backend to attempt. If it's not installed, an ImportError
+            will be raised with a user‐friendly message.
+
+        Returns
+        -------
+        - If `library="matplotlib"`, returns `(fig, ax)`.
+        - If `library="plotly"`, returns a `plotly.graph_objects.Figure`.
+
+        Raises
+        ------
+        ImportError
+            If the requested plotting library cannot be imported.
+        IndexError
+            If `mode` ≥ `state.d`.
+        """
+
+        # 1) Extract ℏ, dimension d, and check `mode`
+        ℏ = float(self._config.hbar)
```

Comment:
Let's refrain from using characters that cannot be accessed easily from a "usual" keyboard for better maintainability and rename `ℏ` to something equally descriptive, e.g., `_hbar`.